File: content/manual/v8.2/source/includes/qe-tutorials/java/SubstringQuery.java
// start-enable-substring
BsonDocument encryptedFieldsMap = new BsonDocument().append("fields",
new BsonArray(Arrays.asList(
        new BsonDocument()
                .append("keyId", new BsonNull())
                .append("path", new BsonString("patientRecord.ssn"))
                .append("bsonType", new BsonString("string"))
                .append("queries", new BsonDocument()
                        .append("queryType", new BsonString("substringPreview"))
                        .append("strMaxLength", new BsonInt32(12))
                        .append("strMinQueryLength", new BsonInt32(3))
                        .append("strMaxQueryLength", new BsonInt32(10))
                        .append("caseSensitive", new BsonBoolean(true))
                        .append("diacriticSensitive", new BsonBoolean(true))))));
// end-enable-substring

// start-query-substring
Document filter = new Document("$expr", 
    new Document("$encStrContains", 
        new Document()
            .append("input", "$patientRecord.ssn")
            .append("substring", "-65-432")));
            
Patient findResult = collection.find(filter).first();

System.out.println(findResult);
// end-query-substring

File: content/manual/v8.2/source/includes/qe-tutorials/csharp/SubstringQuery.cs
// start-enable-substring
var encryptedFields = new BsonDocument
{
    {
        "fields", new BsonArray
        {
            new BsonDocument
            {
                { "keyId", BsonNull.Value },
                { "path", "patientRecord.ssn" },
                { "bsonType", "string" },
                { "queries", new BsonDocument
                    {
                        { "queryType", "substringPreview" },
                        { "strMaxLength", 12 },
                        { "strMinQueryLength", 3 },
                        { "strMaxQueryLength", 10 },
                        { "caseSensitive", true },
                        { "diacriticSensitive", true }
                    }
                }
            }
        }
    }
};
// end-enable-substring

// start-query-substring
var filter = new BsonDocument("$expr", new BsonDocument("$encStrContains", 
    new BsonDocument
    {
        { "input", "$patientRecord.ssn" },
        { "substring", "-65-432" }
    }));

var findResult = await encryptedCollection.Find(filter).ToCursorAsync();
Console.WriteLine(findResult.FirstOrDefault().ToJson());
// end-query-substring